core/index/index_writer.cpp

```diff
@@ -2271,6 +2270,15 @@ IndexWriter::PendingContext IndexWriter::PrepareFlush(const CommitInfo& info) {
 
   // only flush a new index version upon a new index or a metadata change
   if (!modified) {
+    IRS_ASSERT(readers.size() == committed_reader_size);
```

Comment:
Do we have test that covers this change?

Reply:
Assert is just my paranoia.
If you about fix, yes I added test.
If you about reopen_columnstore test in arangodb for now

Reply:
I was about test for "reopen". 
Please add tests to iresearch.

Reply:
Done